### PR TITLE
Scope Frequency Domain Measurement

### PR DESCRIPTION
**Pull request by Jay-Cao123**
### What does this Pull Request accomplish?

- This measurement includes a LabVIEW project and measurement service library that automatically generated by 'Measurement Plug-in'.
- The VIs in the library are updated to support NI-SCOPE Frequency Domain measurements.
- A TestStand sequence file is included to show how to configure the pin map, register instrument sessions with the session management service, and run a measurement.
- InstrumentStudio files are included to show the manual testing.
- A Pin map file is included.
- A readme file is included.

### Why should this Pull Request be merged?

Adding this as one of the PCBA Function Test measurements.

### What testing has been done?

Test setup:

NI PXIe-5122 (SCOPE1) CH1 is connected to NI PXIe-4071 (DAQmx1) AO1.
The output signal of AO1: Voltage sinewave, 2 volts amplitude, 1Hz.
image

Test result via built exe:
InstrumentStudio:
image
TestStand:
image

Build result:
image

### CODE REVIEW

**Review by nathan-murphy — CHANGES_REQUESTED** [review]

File: src/NIScopeFrequencyDomain/NIScopeFrequencyDomain/SubVI/Magnitudes Measurement.vi
[the remark this answers is not in the document]
Comment (kirstyong) [reply src/NIScopeFrequencyDomain/NIScopeFrequencyDomain/SubVI/Magnitudes Measurement.vi:1]: Could you do a quick run through of your logic VI and subVI's and just make sure that wires entering/exiting vi's are aligned with where they are meant to enter/exit visually? ex. below to move bottom wire going into Magnitude RMS to be moved to left a bit

image

**Review by kirstyong — CHANGES_REQUESTED** [review]
Overall looks good! I validated it with a different frequency sinewave (10Hz) from a DAQmx example and can see that is returned with the right amplitude. 👍  Please add a readme and are the screenshots in the PR from testing using the built exe? If so, could you include that in a note in the PR description. Thanks!

image

File: src/NIScopeFrequencyDomain/NIScopeFrequencyDomain/Measurement UI.vi
[the remark this answers is not in the document]
Comment (Jay-Cao123) [reply src/NIScopeFrequencyDomain/NIScopeFrequencyDomain/Measurement UI.vi:1]: So in order to set the x-axis starting as 1Hz, I dis-select the "auto-scale" and make it as :
image

For a 1kHz sinewave, here's the graph:
image
For a 1Hz sinewave, here's the graph:
image
Is this what you want?

[the remark this answers is not in the document]
Comment (nathan-murphy) [reply src/NIScopeFrequencyDomain/NIScopeFrequencyDomain/Measurement UI.vi:1]: Here's how you can convert numeric types into SI notation pretty easily:

image

[the remark this answers is not in the document]
Comment (nathan-murphy) [reply src/NIScopeFrequencyDomain/NIScopeFrequencyDomain/Measurement UI.vi:1]: Thanks for providing screenshots. They look strange with the forced range on the x-axis. I think we can turn autoscaling back on, but definitely leave it in log scale.

**Review by nathan-murphy — APPROVED** [review]

**Review by kena0405 — APPROVED** [review]

**Comment by Jay-Cao123** [comment]
Final UI:
image

File: src/NIScopeFrequencyDomain/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  - For the sake of simplicity, the TestStand sequence handles pin map and session registration and unregistration in the `Setup` and `Cleanup` sections of the main sequence. For **Sequential** and **Batch** process model use cases, these steps should be moved to the `ProcessSetup` and `ProcessCleanup` callbacks.
+- Uses the NI gRPC Device Server to allow sharing instrument sessions with other measurement services when running measurements from TestStand.
+
+### Required Driver Software
+
+- NI-SCOPE 2024 Q1 or later
+
+### Required Software
+
+- InstrumentStudio 2024Q1 or later
```
Comment (kirstyong) [inline src/NIScopeFrequencyDomain/README.md:22]: Small change, can you add a space between 2024 and Q1 for all of these? Thanks!

**Review by kirstyong — APPROVED** [review]

**Pull request merged**